lib/main/atom/atomConfig.ts

```diff
@@ -0,0 +1,31 @@
+import {getName} from "../lang/utils";
+
+interface IConfig extends AtomCore.IConfig {
+  set<T>(keyPath:string, value: T):any;
+}
+// Documentation https://atom.io/docs/api/v0.177.0/Config and http://json-schema.org/examples.html
+// To add a new setting you need to add to
+//    schema
+//    getter/setter
+
+var packageName = 'atom-typescript';
+function getConfig<T>(nameLambda: () => any): T {
+    return atom.config.get(packageName + '.' + getName(nameLambda));
+}
+function setConfig<T>(nameLambda: () => any, value: T): T {
+    return (atom.config as IConfig).set(packageName + '.' + getName(nameLambda), value);
+}
+
+class Config {
+    schema = {
+        showSemanticView: {
```

Comment:
The semantic view probably doesn't need a setting if we can make it non-intrusive and easy to toggle when necessary.